pkg/operation/botanist/controlplane.go

```diff
@@ -777,3 +747,138 @@ func (b *Botanist) HibernateControlPlane(ctx context.Context) error {
 
 	return nil
 }
+
+// ControlPlaneDefaultTimeout is the default timeout and defines how long Gardener should wait
+// for a successful reconciliation of a control plane resource.
+const ControlPlaneDefaultTimeout = 10 * time.Minute
+
+// DeployControlPlane creates the `ControlPlane` extension resource in the shoot namespace in the seed
+// cluster. Gardener waits until an external controller did reconcile the cluster successfully.
+func (b *Botanist) DeployControlPlane(ctx context.Context) error {
+	var (
+		cp = &extensionsv1alpha1.ControlPlane{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      b.Shoot.Info.Name,
+				Namespace: b.Shoot.SeedNamespace,
+			},
+		}
+	)
+
+	// In the future the providerConfig will be blindly copied from the core.gardener.cloud/v1alpha1.Shoot
+	// resource. However, until we have completely moved to this resource, we have to compute the needed
+	// configuration ourselves from garden.sapcloud.io/v1beta1.Shoot.
+	providerConfig, err := migration.ShootToControlPlaneConfig(b.Shoot.Info)
+	if err != nil {
+		return err
+	}
+
+	return kutil.CreateOrUpdate(ctx, b.K8sSeedClient.Client(), cp, func() error {
+		cp.Spec = extensionsv1alpha1.ControlPlaneSpec{
+			DefaultSpec: extensionsv1alpha1.DefaultSpec{
+				Type: string(b.Shoot.CloudProvider),
+			},
+			Region: b.Shoot.Info.Spec.Cloud.Region,
+			SecretRef: corev1.SecretReference{
+				Name:      gardencorev1alpha1.SecretNameCloudProvider,
+				Namespace: cp.Namespace,
+			},
+			ProviderConfig: &runtime.RawExtension{
+				Object: providerConfig,
+			},
+			InfrastructureProviderStatus: &runtime.RawExtension{
+				Raw: b.Shoot.InfrastructureStatus,
+			},
+		}
+		return nil
+	})
+}
+
+// DestroyControlPlane deletes the `ControlPlane` extension resource in the shoot namespace in the seed cluster,
+// and it waits for a maximum of 10m until it is deleted.
+func (b *Botanist) DestroyControlPlane(ctx context.Context) error {
+	if err := b.K8sSeedClient.Client().Delete(ctx, &extensionsv1alpha1.ControlPlane{ObjectMeta: metav1.ObjectMeta{Namespace: b.Shoot.SeedNamespace, Name: b.Shoot.Info.Name}}); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
```suggestion
	return client.IgnoreNotFound(b.K8sSeedClient.Client().Delete(ctx, &extensionsv1alpha1.ControlPlane{ObjectMeta: metav1.ObjectMeta{Namespace: b.Shoot.SeedNamespace, Name: b.Shoot.Info.Name}}))
```

Reply:
The 2 are not equivalent (due to `!apierrors.IsNotFound(err)`), I think the code should stay as it is

Reply:
Why are they not equivalent? Can you explain again? The `client.IgnoreNotFound` function is returning `nil` if `apierrors.IsNotFound(err) == true`.